plugins/PreviewStage/PreviewMenu.qml

```diff
@@ -22,135 +22,122 @@ Item
 
-    Rectangle
-    {
-        anchors.fill: stageMenu
-        anchors.leftMargin: -radius
-        radius: UM.Theme.getSize("default_radius").width
-        color: UM.Theme.getColor("toolbar_background")
-    }
 
-    Item
+    Row
     {
-        id: stageMenu
+        id: stageMenuRow
+        anchors.centerIn: parent
         height: parent.height
-        width: stageMenuRow.width + UM.Theme.getSize("default_margin").width
-        anchors.horizontalCenter: parent.horizontalCenter
-        Row
+
+        Cura.ExpandableComponent
         {
-            id: stageMenuRow
-            anchors.centerIn: parent
+            id: viewSelector
+            iconSource: expanded ? UM.Theme.getIcon("arrow_bottom") : UM.Theme.getIcon("arrow_left")
             height: parent.height
+            headerCornerSide: 2 // Show corners on the left side
 
-            Cura.ExpandableComponent
-            {
-                id: viewSelector
-                iconSource: expanded ? UM.Theme.getIcon("arrow_bottom") : UM.Theme.getIcon("arrow_left")
-                height: parent.height
-
-                property var viewModel: UM.ViewModel { }
+            property var viewModel: UM.ViewModel { }
 
-                property var activeView:
+            property var activeView:
+            {
+                for (var i = 0; i < viewModel.rowCount(); i++)
                 {
-                    for (var i = 0; i < viewModel.rowCount(); i++)
+                    if (viewModel.items[i].active)
                     {
-                        if (viewModel.items[i].active)
-                        {
-                            return viewModel.items[i]
-                        }
+                        return viewModel.items[i]
                     }
-                    return null
                 }
+                return null
+            }
 
-                Component.onCompleted:
+            Component.onCompleted:
+            {
+                // Nothing was active, so just return the first one (the list is sorted by priority, so the most
+                // important one should be returned)
+                if(activeView == null)
                 {
-                    // Nothing was active, so just return the first one (the list is sorted by priority, so the most
-                    // important one should be returned)
-                    if(activeView == null)
-                    {
-                        UM.Controller.setActiveView(viewModel.getItem(0).id)
-                    }
+                    UM.Controller.setActiveView(viewModel.getItem(0).id)
                 }
+            }
+
+            headerItem: Label
+            {
+                text: viewSelector.activeView ? viewSelector.activeView.name : ""
+                verticalAlignment: Text.AlignVCenter
+                height: parent.height
+                elide: Text.ElideRight
+                font: UM.Theme.getFont("default")
+                color: UM.Theme.getColor("text")
```

Comment:
Use `renderType: Text.NativeRendering` otherwise the texts in some OS look weird.